10-setup-htcondor.sh

```diff
@@ -228,23 +219,64 @@ EOF
 
 fi
 
-if is_true "$ITBPOOL"; then
-    OSPOOL_CM1=cm-1.ospool-itb.osg-htc.org
-    OSPOOL_CM2=cm-2.ospool-itb.osg-htc.org
+
+######################
+# POOL CONFIGURATION #
+######################
+
+# Default to the production OSPool
+POOL=${POOL:=prod-ospool}
+
+case ${POOL} in
+    itb-ospool)
+        CCB_RANGE_LOW=1
+        CCB_RANGE_HIGH=5
+        default_cm1=cm-1.ospool-itb.osg-htc.org
+        default_cm2=cm-2.ospool-itb.osg-htc.org
+        default_syslog_host=syslog.osgdev.chtc.io
+        ;;
+    prod-ospool)
+        CCB_RANGE_LOW=1
+        CCB_RANGE_HIGH=5
+        default_cm1=cm-1.ospool.osg-htc.org
+        default_cm2=cm-2.ospool.osg-htc.org
+        default_syslog_host=syslog.osg.chtc.io
+        ;;
+    prod-path-facility)
+        CCB_RANGE_LOW=
+        CCB_RANGE_HIGH=
+        default_cm1=cm-1.facility.path-cc.io
+        default_cm2=cm-2.facility.path-cc.io
+        default_syslog_host=syslog.osg.chtc.io
+        ;;
+    *)
+        echo "Unknown pool $POOL" >&2
+        exit 1
+        ;;
+esac
+
+# Allow users to override the pool configuration
+if [[ -n $CONDOR_HOST ]]; then
+    # if the user sets $CONDOR_HOST, we can't assume the POOL
+    # so we unset it here to avoid confusion
+    POOL=
 else
-    OSPOOL_CM1=cm-1.ospool.osg-htc.org
-    OSPOOL_CM2=cm-2.ospool.osg-htc.org
+    CONDOR_HOST=$default_cm1,$default_cm2
 fi
 
-# extra HTCondor config
-# if CCB_RANGE_* is set, use the old config, otherwise assume OSPool with shared port
-if [[ "x$CCB_RANGE_LOW" != "x" ]]; then
+# Configure remote peer if applicable
+SYSLOG_HOST=${SYSLOG_HOST:-$default_syslog_host}
+
+
+# Don't assume a CCB unless they specify CCB_RANGE_LOW or CCB_RANGE_HIGH
+if [[ -n $CCB_RANGE_LOW ]] || [[ -n $CCB_RANGE_HIGH ]]; then
     CCB_PORT=$(python -S -c "import random; print(random.randrange($CCB_RANGE_LOW,$CCB_RANGE_HIGH+1))")
-    CCB_ADDRESS="\$(CONDOR_HOST):$CCB_PORT"
-else
-    CCB_COLLECTOR=$(python -S -c "import random; print(random.randrange(1,6))")
-    CCB_ADDRESS="${OSPOOL_CM1}:9619?sock=collector$CCB_COLLECTOR,${OSPOOL_CM2}:9619?sock=collector$CCB_COLLECTOR"
+    if [[ $POOL =~ (itb|prod)-ospool ]]; then
+        CCB_PORT="9619?sock=collector$CCB_PORT"
```

Comment:
Whatever this is, it is not a port.

I'm really having trouble following this code or figuring out what comes out of it. IMO it's OK to repeat yourself if it enhances clarity.

Reply:
Does the most recent commit make it clearer?

Reply:
Not really, and it was more readable when you set the CCB low/high in the `case` statement.

Perhaps you could add an example of what the string is expected to look like?

Reply:
Added and just reworked the tortured logic. Should be clearer now.